pkg/destroy/gcp/network.go

```diff
@@ -0,0 +1,647 @@
+		if len(item.Backends) == 0 {
+			return false
+		}
+		for _, backend := range item.Backends {
+			if !urls.Has(backend.Group) {
+				return false
+			}
+		}
+		return true
+	})
+}
+
+// deleteInternalLoadBalancer follows a similar cleanup procedure as:
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go#L222
+// TODO: add cleanup for shared mode resources (determine if it's supported in 4.2)
+func (o *ClusterUninstaller) deleteInternalLoadBalancer(clusterID string, loadBalancerName string) error {
+	if err := o.deleteAddress(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteForwardingRule(loadBalancerName); err != nil {
+		return err
+	}
+	// TODO: Figure out a way to preserve the backend service name after this is
+	// gone. Otherwise we can't find this internal load balancer again. However,
+	// the backend service must be deleted first before health checks.
+	if err := o.deleteBackendService(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteHealthCheck(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteHealthCheck(loadBalancerName + "-hc"); err != nil {
+		return err
+	}
+	if err := o.deleteFirewall(loadBalancerName + "-hc"); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteExternalLoadBalancer follows a similar cleanup procedure as:
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go#L289
+// TODO: cleanup nodes health check (using clusterid)
+func (o *ClusterUninstaller) deleteExternalLoadBalancer(loadBalancerName string) error {
+	if err := o.deleteAddress(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteForwardingRule(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteFirewall(fmt.Sprintf("k8s-fw-%s", loadBalancerName)); err != nil {
+		return err
+	}
+	if err := o.deleteTargetPool(loadBalancerName); err != nil {
+		return err
+	}
+	if err := o.deleteHealthCheck(loadBalancerName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *ClusterUninstaller) destroyCloudControllerInternalLBs() error {
+	groups, err := o.getInternalLBInstanceGroups()
+	if err != nil {
+		return err
+	}
+	if len(groups) == 0 {
+		return nil
+	}
+	clusterID := strings.TrimPrefix(groups[0].name, "k8s-ig--")
+	backends, err := o.listBackendServicesForInstanceGroups(groups)
+	if err != nil {
+		return err
+	}
+
+	// Each backend found represents an internal load balancer.
+	// For each, remove related resources
+	for _, backend := range backends {
+		err := o.deleteInternalLoadBalancer(clusterID, backend)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Finally, remove the instance groups {
+	for _, group := range groups {
+		err := o.deleteComputeInstance(group)
```

Comment:
this should be `deleteInstanceGroup` right?

Reply:
yup